### DISCUSSION TITLE
How does the server deliver pictures

### DISCUSSION THREAD

**Discussion by Zhangbecool** [FreeOpcUa/python-opcua#1427] [Q&A] [answered]
I added an object and I want to add a picture type variable but I don't know how to submit the picture

**Comment by schroeder-** [comment, marked as answer]
Picture Type is abstract, you have to use ImageBMP, ImageGIF, ImageJPG or ImagePNG. They are just ByteStrings => Python bytes.
`image = b'0x123455'`

**Reply by Zhangbecool** [reply]
`self.img = self.myobj.add_variable(idx, 'image', None, datatype=ua.ObjectIds.ImageJPG)
        with open('1.jpg', 'rb') as f:
            encode_img_info = base64.b64encode(f.read())
        self.img.set_value(encode_img_info)`


Is that what you mean

**Reply by oroulet** [reply]
no need for encoding, write the bytes directy

**Reply by Zhangbecool** [reply]
thank you it's working

**Comment by oroulet** [comment]
so read_value() and write_value() on the node

**Reply by Zhangbecool** [reply]
I mean, how to transfer pictures on the server side

**Reply by AndreasHeine** [reply]
transfer is client part!
